examples/scripts/sft_trainer.py

```diff
@@ -99,20 +100,24 @@ class ScriptArguments:
     model=model,
     args=training_args,
     max_seq_length=script_args.seq_length,
-    train_dataset=dataset,
+    train_dataset=train_dataset,
+    eval_dataset=eval_dataset,
     dataset_text_field=script_args.dataset_text_field,
     peft_config=peft_config,
 )
 
 trainer.add_callback(TBTrainerCallback())
 
 train_result = trainer.train()
+eval_result = trainer.evaluate()
 
 # Step 6: Save the model
 trainer.save_model()
 
 trainer.log_metrics("train", train_result.metrics)
 trainer.save_metrics("train", train_result.metrics)
 
+trainer.log_metrics("eval", eval_result.metrics)
```

Comment:
What kind of metrics is it expected to have? I thought we are going to use `train_loss` for key_metric

Reply:
fixed this